Sources/LiveKit/Extensions/RTCVideoCapturerDelegate+Buffer.swift

```diff
@@ -1,6 +1,20 @@
 import WebRTC
 import ReplayKit
 
+extension Dimensions {
+
+    // Ensures width and height are even numbers
+    internal func toEncodeSafeDimensions() -> Dimensions {
+        Dimensions(width: max(encodeSafeSize, width % 2 == 0 ? width : width + 1),
+                   height: max(encodeSafeSize, height % 2 == 0 ? height : height + 1))
```

Comment:
Maybe better to reduce one pixel as one pixel of extra blank could be visible and make it look odd.

Reply:
On second thoughts wondering if this should also round down to multiple of 8 to be safe. Have you tried non-multiple of 8?

Reply:
I think the following code should stretch the frame before passing it to the encoder (if required), so there are no black 1 px lines.

It's definitely worth experimenting with multiples of 8, but as of now I haven't encountered a single crash with multiple of 2. 🙂

```swift
            if sourceDimensions != targetDimensions {
                // no adjustments required
                rtcBuffer = RTCCVPixelBuffer(pixelBuffer: pixelBuffer)
            } else {
                // apply adjustments
                rtcBuffer = RTCCVPixelBuffer(pixelBuffer: pixelBuffer,
                                             adaptedWidth: targetDimensions.width,
                                             adaptedHeight: targetDimensions.height,
                                             cropWidth: sourceDimensions.width,
                                             cropHeight: sourceDimensions.height,
                                             cropX: 0,
                                             cropY: 0)
            }

```

Reply:
After testing, it feel like power of 8 or even 4 seems to work better 🙂
Is it ok to roundUp vs roundDown ?

Reply:
If you are scaling, I would suggest keeping the round up/round down to a minimum. If mod 2 works and if you have round up or round down by 1, that is better because scaling will not introduce any artefacts. If you are rounding up/down by 4/8 and scaling, the image may be stretched and may look a bit off compared to the capture as it is stretching/shrinking by more.

Reply:
I was thinking about cropping. If you are cropping, it is better to round down to lose one pixel rather than add a pixel and have that be a black/grey/white line which will stand out.

Reply:
Thanks for the info 😁🙏🏻 i will do more experimenting 🙂